.github/workflows/build_and_test.yml

```diff
@@ -0,0 +1,256 @@
+name: Build and test
+
+on:
+  push:
+    branches:
+      - master
+      - "*-rc"
+  pull_request:
+    types:
+      - opened
+      - synchronize
+      - reopened
+    branches:
+      - "**"
+
+jobs:
+  verify_files:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Setup
+        run: |
+          sudo apt-get update -y && sudo apt-get install -y gnupg2
+
+      - name: Verify files
+        run: |
+          curl -sSL https://secchannel.rsk.co/SUPPORT.asc | gpg --import -
+          gpg --verify SHA256SUMS.asc && sha256sum --check SHA256SUMS.asc
+
+  clone_rskj_repo:
+    needs: verify_files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout RSKj repo
+        uses: actions/checkout@v4
+        with:
+          repository: rsksmart/rskj
+          token: ${{ secrets.GITHUB_TOKEN }}
+          path: rskj
+          fetch-depth: 0
+      
+      - name: Determine branch to checkout for PR
+        if: github.event_name == 'pull_request'
+        working-directory: rskj
+        env:
+          PR_BRANCH: ${{ github.head_ref }}
+        run: |
+          IS_RSKJ_BRANCH=`git ls-remote --heads origin $PR_BRANCH`
```

Comment:
ugh, can't believe I missed the `PR_BRANCH` definition just before it :(

I meant to quote the `$PR_BRANCH` use, not the definition. In the definition it makes no difference, because it's YAML. In its use, `git ls-remote --heads origin $PR_BRANCH` will do funny things if it `$PR_BRANCH` isn't quoted and, for example, has spaces.

Except in the case that you fully control the value of variables and you know that in all cases it doesn't include characters from the input field separators, you should quote the variable usage in shell scripts.

Reply:
Thanks for pointing that out! I've updated the workflow to quote both PR_BRANCH and POW_REF. They both look something like this:

```
env:
  PR_BRANCH: ${{ github.head_ref }}
run: |
  IS_RSKJ_BRANCH=$(git ls-remote --heads origin "$PR_BRANCH")
```
Appreciate the help in catching that!